pkg/storage/vault/aws/aws.go

```diff
@@ -0,0 +1,104 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsConfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/mitchellh/mapstructure"
+)
+
+type AWSVault struct {
+	Client          *secretsmanager.Client
+	AccessKeyId     string `mapstructure:"access_key_id"`
+	SecretAccessKey string `mapstructure:"secret_access_key"`
+	Prefix          string `mapstructure:"prefix"`
+	Region          string `mapstructure:"region"`
+}
+
+func NewAWSVault(conf map[string]any) (*AWSVault, error) {
+	var vault AWSVault
+
+	err := mapstructure.Decode(conf, &vault)
+	if err != nil {
+		return nil, err
+	}
+
+	if vault.AccessKeyId == "" {
+		return nil, errors.New("AccessKeyId not found or not a string")
+	}
+
+	if vault.SecretAccessKey == "" {
+		return nil, errors.New("SecretAccessKey not found or not a string")
+	}
+
+	if vault.Prefix == "" {
```

Comment:
A prefix isn't required if the user doesn't want to specify one. (I know this was copied from the previous PR.)